index.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield self[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    array = []
+    return to_enum unless block_given?
+
+    0.upto(length - 1) do |i|
+      array << self[i] if yield self[i]
+    end
+    array
+  end
+
+  def my_all?(patt = nil)
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif patt.nil?
+      my_each { |i| return false unless i }
+    else
+      my_each { |i| return false unless check_patt(i, patt) }
+    end
+    true
+  end
+
+  def my_none?(arg = nil)
+    !my_all?(arg)
+  end
+
+  def my_any?(patt = nil)
+    if block_given?
+      my_each { |i| return true unless yield(i) }
+    elsif patt.nil?
+      my_each { |i| return true unless i }
+    else
+      my_each { |i| return true unless check_patt(i, patt) }
+    end
+    false
```

Comment:
* [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5,6].my_any?{ |num| num <= 6 } != [1,2,3,4,5,6].any?{ |num| num <= 6 }

[nil, false, nil, false].my_any? != [nil, false, nil, false].any?

[1,2,3,4,5].my_any?(Integer) != [1,2,3,4,5].any?(Integer)

%w[dog door rod blade].my_any?(Integer) != [dog door rod blade].any?(Integer)

%w[dog door rod blade].my_any?(/z/) != [dog door rod blade].any?(/z/)

%w[dog door rod blade].my_any?('cat') != %w[dog door rod blade].any?('cat')
```